lib/rules/indent.js

```diff
@@ -807,6 +889,84 @@ module.exports = {
             });
         }
 
+        /**
+         * Iterate through the dependent tokens until we reach the last token
+         * dependency or the last token dependency that is within the unknown
+         * node. If the last token dependency is not the first token of the line
+         * make the last token dependcy depend on the first token of the line.
+         * This ensures the offset will be set correctly since only the first
+         * token of each line is ignored and not all tokens on a line.
+         * @param {Token} dependency First
+         * @param {number} startLineOfNode First line number of unknown node
+         * @returns {void}
+         */
+        function ignoreUnknownNodeLineWithDependency(dependency, startLineOfNode) {
+            let currDependency = dependency,
+                lastDependency = dependency;
+
+            while (
+                (currDependency = offsets.getFirstDependency(currDependency)) &&
+                currDependency.loc.start.line > startLineOfNode
```

Comment:
This is an interesting idea and will work in some cases. It does not work when we have token dependencies that are not on the first line.

This change will make all tokens depend on the first token of the first line of the unknown node. For example:
```ts
1 abstract class X {
2     public baz() {
3         if (true) {
4             qux();
5         }
6     }
7 }
```

Line 3 offset depends on Line 2 tokens `{` and `(`, which depend on Line 1 token `abstract`.

Since we only ignore the first token of each line, only the token `public` on Line 2 is ignored. This means the offset for Line 2 tokens `{` and `(` is 0 when it should be 1. The modification we made to `getDesiredIndent()`, to use the offset of the first token of the line instead of 0, will only work wen there is no further dependencies. I tried to modify `getDesiredIndent()` to always get the offset from the first token of the line, even when from is set, but this causes the indent rule to break. 

I also tried to modify `getDesiredIndent()` to use the offset of the first token when it is ignored. This would have the same effect of removing the dependency from every token of the line when it is ignored. But it caused some tests to fail.

Any other possible solutions? I can push the working solution I have currently. This includes modifying `getDesiredIndent()` to use the first token of a line when there is no token dependency and removing the token dependencies from every token of a line only when the dependency falls outside of the unknown node.